src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/Macros/ProcessValueFormMacro.cs

```diff
@@ -43,11 +43,21 @@ public void EvaluateConfig(IEngineEnvironmentSettings environmentSettings, IVari
             {
                 value = form.Process(realConfig.Forms, value);
 
-                Parameter p = new Parameter
+                Parameter p;
+
+                if (parameters.TryGetParameterDefinition(config.VariableName, out ITemplateParameter existingParam))
+                {   // Derived parameters already have a definition, but need value form processing.
+                    // When the param already exists, use its definition.
+                    p = (Parameter)existingParam;
```

Comment:
Let's set `IsVariable = true` on the existing parameter for consistency (even though it's never used right now) and make all of the other macros behave this way too.

We should probably make this comment a little more clear as well (since we're in the value forms macro) that the existing parameter must be reused to allow the processing of value forms declared on the derived symbol type later in the thing that deals with value forms for parameter type symbols.

We should also make sure that the processing of the value forms macro for the derived symbol happens before value forms for parameters/derived symbols happen